webapp/src/Components/Replay/ReplayViewer/ReplayViewer.tsx

```diff
@@ -32,16 +40,92 @@ export class ReplayViewer extends React.PureComponent<Props, State> {
                         Replay Viewer
                     </Typography>
                 </Grid>
-                <Grid item xs={12}>
-                    {this.state.replayData &&
-                    <ThreeScene replayData={this.state.replayData}/>}
+                <Grid item xs={8}>
+                    {
+                        !this.state.replayData &&
+                        <Typography>Loading...</Typography>
+                    }
+                    {
+                        this.state.replayData &&
+                        <ThreeScene
+                            replayData={this.state.replayData}
+                            frame={this.state.currentFrame}
+                        />
+                    }
+                </Grid>
+                <Grid item xs={4}>
+                    <Typography>Playback Controls</Typography>
+                    <button onClick={this.startPlayback}>Play</button>
+                    <button onClick={this.stopPlayback}>Pause</button>
+                    <Typography>
+                        {/*We should use padStart on seconds, but that is in es2017, not es6*/}
+                        Game Time: {(this.state.gameTime / 60).toFixed(0)}:{(this.state.gameTime % 60).toString()}
+                    </Typography>
+                </Grid>
+                <Grid item xs={6}>
+                    <label>Frame:</label>
+                    <input type="number" value={this.state.currentFrame} onChange={this.setCurrentFrame}/>
+                </Grid>
+                <Grid item xs={6}>
+                    <Typography>
+                        Ball Position:
+                        {
+                            this.state.replayData &&
+                            this.state.replayData.ball[this.state.currentFrame][0]
+                        },
+                        {
+                            this.state.replayData &&
+                            this.state.replayData.ball[this.state.currentFrame][1]
+                        },
+                        {
+                            this.state.replayData &&
+                            this.state.replayData.ball[this.state.currentFrame][2]
+                        }
+                    </Typography>
                 </Grid>
             </Grid>
         )
     }
 
-    private readonly getReplayPositions = (): Promise<any> => {
-        return getReplayViewerData(this.props.replay.id)
-            .then((data: any) => this.setState({replayData: data}))
+    private readonly getReplayPositions = async() => {
+        const data: any = await getReplayViewerData(this.props.replay.id)
+        this.setState({replayData: data})
+    }
+
+    private readonly setCurrentFrame: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+        const value: number = parseInt(event.target.value, 10)
+        this.setState({currentFrame: value})
+        this.updateeGameTime()
+    }
+
+    private updateeGameTime = () => {
```

Comment:
can we actually separate out the updating and playback options into a separate file?

We will eventually want to use the exact same viewer code but for things like viewing shots.  where the controls and the format will be very different.